govcd/vapp.go

```diff
@@ -814,7 +1148,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
 	return nil
 }
 
+// RemoveNetwork removes any network (be it isolated or connected to an Org Network) from vApp
+// Returns pointer to types.NetworkConfigSection or error
+func (vapp *VApp) RemoveNetwork(id string) (*types.NetworkConfigSection, error) {
+	task, err := vapp.RemoveNetworkAsync(id)
+	if err != nil {
+		return nil, err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
+	}
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// RemoveNetworkAsync asyncronously removes any network (be it isolated or connected to an Org Network) from vApp
+// Accepts network ID or name
+func (vapp *VApp) RemoveNetworkAsync(id string) (Task, error) {
```

Comment:
I believe we use such variable name when name and id are accepted.

```suggestion
func (vapp *VApp) RemoveNetworkAsync(identifier string) (Task, error) {
```

Reply:
changed